Src/FluentAssertions/AndWhichConstraint.cs

```diff
@@ -2,45 +2,84 @@
+    /// </summary>
+    public AndWhichConstraint(TParent parent, TSubject subject)
+        : base(parent)
+    {
+        getSubject = new Lazy<TSubject>(() => subject);
+    }
+
+    /// <summary>
+    /// Creates an object that allows continuing an assertion executed through <paramref name="parent"/> and
+    /// which resulted in a single <paramref name="subject"/> on an existing <paramref name="assertionChain"/>, but where
+    /// the previous caller identifier is post-fixed with <paramref name="pathPostfix"/>.
+    /// </summary>
+    public AndWhichConstraint(TParent parent, TSubject subject, AssertionChain assertionChain, string pathPostfix = "")
+        : base(parent)
     {
-        this.matchedConstraint =
-            new Lazy<TMatchedElement>(() => matchedConstraint);
+        getSubject = new Lazy<TSubject>(() => subject);
+
+        this.assertionChain = assertionChain;
+        this.pathPostfix = pathPostfix;
     }
 
-    public AndWhichConstraint(TParentConstraint parentConstraint, IEnumerable<TMatchedElement> matchedConstraint)
-        : base(parentConstraint)
+    /// <summary>
+    /// Creates an object that allows continuing an assertion executed through <paramref name="parent"/> and
+    /// which resulted in a potential collection of objects through <paramref name="subjects"/>.
+    /// </summary>
+    /// <remarks>
+    /// If <paramref name="subjects"/> contains more than one object, a clear exception is thrown.
+    /// </remarks>
+    public AndWhichConstraint(TParent parent, IEnumerable<TSubject> subjects)
+        : base(parent)
     {
-        this.matchedConstraint =
-            new Lazy<TMatchedElement>(
-                () => SingleOrDefault(matchedConstraint));
+        getSubject = new Lazy<TSubject>(() => SingleOrDefault(subjects));
     }
 
-    private static TMatchedElement SingleOrDefault(
-        IEnumerable<TMatchedElement> matchedConstraint)
+    /// <summary>
+    /// Creates an object that allows continuing an assertion executed through <paramref name="parent"/> and
+    /// which resulted in a potential collection of objects through <paramref name="subjects"/> on an
+    /// existing <paramref name="assertionChain"/>, but where
+    /// the previous caller identifier is post-fixed with <paramref name="pathPostfix"/>.
+    /// </summary>
+    /// <remarks>
+    /// If <paramref name="subjects"/> contains more than one object, a clear exception is thrown.
+    /// </remarks>
+    public AndWhichConstraint(TParent parent, IEnumerable<TSubject> subjects, AssertionChain assertionChain, string pathPostfix)
+        : base(parent)
     {
-        TMatchedElement[] matchedElements = matchedConstraint.ToArray();
+        getSubject = new Lazy<TSubject>(() => SingleOrDefault(subjects));
+
+        this.assertionChain = assertionChain;
+        this.pathPostfix = pathPostfix;
+    }
+
+    private static TSubject SingleOrDefault(IEnumerable<TSubject> subjects)
```

Comment:
🤔 I wonder if this should be named `SingleOrThrow`, becasue there is no `Default` returned like in LINQs `SingleOrDefault`
Or return the first item if >1 are found?

Reply:
I agree.

Reply:
> I agree.

For `SingleOrThrow` or `SingleOrFirst`?

Reply:
Actually, I now named it `Single`, since `Single` on collections will also throw if the collection contains more than one item.